### PR TITLE
explain that non 2xx means no dns answer

### LINKED ISSUE #85: clarify http errors vs dns errors
largely a restating of mailing list thread: A question on the mix of DNS and HTTP semantics

>  I'm certainly happy to add a sentence or two to help make that more clear (suggestions?) but I don't want to start defining/reiterating HTTP semantics in the DoH doc - we've already got lots of documents (i.e. rfc723x.. which are all included normatively by 7230 which doh includes normatively) that do that and its not in doh's charter to mess those things anyhow.

### PR DESCRIPTION
**Pull request by mcmanus**
closes #85

### CODE REVIEW

File: draft-ietf-doh-dns-over-https-latest.mkd [hunk 1]
```diff
@@ -311,6 +312,16 @@ accept = application/dns-udpwireformat
 
 # The HTTP Response
 
+A successful DOH response will carry DNS information in the HTTP
+response with a status code between 200 and 299. A successful HTTP
+response indicates a valid DNS response to the query made in the
+HTTP request but that DNS level response itself might represent some
+sort of failure (e.g. NXDOMAIN). HTTP response codes that are not in
```
Comment (martinthomson) [inline draft-ietf-doh-dns-over-https-latest.mkd]: The connection between "successful DOH response" and 2xx isn't concrete enough here.

An HTTP response with a 2xx status code indicates a valid DNS response to the query made in the HTTP request.  A valid DNS response includes both success and failure responses.

**Comment by mcmanus** [comment]
Thats better. Thanks

**Comment by vttale** [comment]
It would be good to explicitly call out the DNS rcode SERVFAIL as a type of failure response as well, not just NXDOMAIN, as they are very different sorts of failures such that the latter can be considered, in some contexts, as not really a failure.

**Comment by mcmanus** [comment]
the merged patch https://github.com/dohwg/draft-ietf-doh-dns-over-https/pull/86/files no longer mentions specific rcodes at all.. wdyt?

**Comment by vttale** [comment]
I personally think that as a DNS implementer that there is enough ambiguity around understanding a valid DNS response, a negative answer, and a failure response, that it should be made more painfully clear that something like all DNS servers returning message-valid SERVFAILs still warrants an HTTP 2xx.

**Comment by paulehoffman** [comment]
Agree with @vttale .

**Comment by mcmanus** [comment]
will reopen the issue to add more text

**Pull request merged**